### ISSUE TITLE
A bit vague but I just had a production version of 1.8.x on Pi 2 (Jessie) crash unrecoverably

### ISSUE THREAD

**Issue by bdickenson** [openhab/openhab#4315]
after multiple restarts etc, result remained the same.  OH 1.8.x refused to start, referencing a log in /var/lib/openhab/workspace/{numbered}.log

All of these logs were "NoClassDefFound: org/eclipse/osgi/internal/baseadaptor/BasePermissionStorage " type --indicating that the underlying OSGI component was missing.   

I eventually recovered by apt-get remove/apt-get install of OH-runtime and my add-ons.  Luckily (:-) I have all my config info in a private github repo)

I say 1.8.x in header because it is possible that 1.8.2 got installed as a part of my recovery efforts -- this production installation has not been deliberately changed since I did a clean install of Jessie on my Pi2 a few months back.

**Comment by bdickenson** [comment]
file:///var/lib/openhab/workspace/1461543170671.log

**Comment by bdickenson** [comment]
in case that link does not work --here is text of the log...


!SESSION 2016-04-24 20:12:49.659 -----------------------------------------------
eclipse.buildId=unknown
java.version=1.8.0_65
java.vendor=Oracle Corporation
BootLoader constants: OS=linux, ARCH=arm, WS=gtk, NL=en_US
Command-line arguments:  -data /var/lib/openhab/workspace -console 5555

!ENTRY org.eclipse.osgi 4 0 2016-04-24 20:12:51.525
!MESSAGE Startup error
!STACK 1
java.lang.NoClassDefFoundError: org/eclipse/osgi/internal/baseadaptor/BasePermissionStorage
	at org.eclipse.osgi.internal.baseadaptor.BaseStorage.readPermissionData(BaseStorage.java:469)
	at org.eclipse.osgi.internal.baseadaptor.BaseStorage.getPermissionStorage(BaseStorage.java:683)
	at org.eclipse.osgi.baseadaptor.BaseAdaptor.getPermissionStorage(BaseAdaptor.java:239)
	at org.eclipse.osgi.framework.internal.core.Framework.<init>(Framework.java:207)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.startup(EclipseStarter.java:262)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:176)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:629)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:584)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1438)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1414)
Caused by: java.lang.ClassNotFoundException: org.eclipse.osgi.internal.baseadaptor.BasePermissionStorage
	at java.net.URLClassLoader$1.run(URLClassLoader.java:370)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 14 more
Caused by: java.util.zip.ZipException: invalid distance too far back
	at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:164)
	at java.util.jar.JarVerifier$VerifierStream.read(JarVerifier.java:478)
	at sun.misc.Resource.getBytes(Resource.java:124)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:462)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	... 19 more

**Comment by watou** [comment]
> Caused by: java.util.zip.ZipException: invalid distance too far back
at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:164)
at java.util.jar.JarVerifier$VerifierStream.read(JarVerifier.java:478)...

This looks like a corrupted JAR file in your installation.  Possible causes are: failing SD card, files corrupted on installation, processor overclocked too much.  If this issue ends up looking like an installation package is corrupted at the origin and can't be installed on a known good system, please reopen this issue.  Consider opening a new discussion topic at https://community.openhab.org.

**Comment by Zefiro** [comment]
I had the same.

My Pi died recently by refusing to boot due to a corrupted file system.
I did a clean format & re-install with everything, but on the same SD card. When it came to installing openhab, I got the above ZipException on startup.

And this is about the only place google finds about this message, the other one being from a different project, but also indicating a broken archive.

Re-installing openhab once again (`apt-get install --reinstall openhab2`) solved this issue. 
Guess I'll better get a new SD card, though :)
